native_client/client.cc

```diff
@@ -305,15 +305,24 @@ WordsFromMetadata(Metadata* metadata)
 }
 
 char* 
-CSVOutput(std::vector<meta_word> words)
+JSONOutput(Metadata* metadata)
 {
+  std::vector<meta_word> words = WordsFromMetadata(metadata);
+
   std::ostringstream out_string;
+  out_string << "{\"metadata\":{\"confidence\":" << std::to_string(metadata->probability) << "},\"words\":[";
```

Comment:
Are these `std::to_string` calls needed? C++ output streams usually have overloads for the basic types.

Reply:
Also, you can use raw string literals to avoid having to escape the quotes:

```cpp
  out_string << R"({"metadata":{"confidence":)" << metadata-probability << R"(},"words":[)";
```